### PR TITLE
Chat: Properly typecheck Chat.runHandlers arguments

### PR DESCRIPTION
**Pull request by mia-pi-git**
This means we shouldn't have more unpleasant issues with changing the types of handlers & missing old uses/runHandler calls.

### CODE REVIEW

File: server/chat.ts [hunk 1]
```diff
@@ -65,15 +65,22 @@ export interface AnnotatedChatCommands {
...
+	onRoomClose: (id: string, user: User, connection: Connection, page: boolean) => any;
+	onRenameRoom: (oldId: RoomID, newID: RoomID, room: BasicRoom) => void;
+	onBattleStart: (user: User, room: GameRoom) => void;
+	onBattleLeave: (user: User, room: GameRoom) => void;
+	onRoomJoin: (room: BasicRoom, user: User, connection: Connection) => void;
+	onDisconnect: (user: User) => void;
+	onRoomDestroy: (roomid: RoomID) => void;
+	onBattleEnd: (battle: RoomBattle, winner: ID, players: ID[]) => void;
+	onLadderSearch: (user: User, connection: Connection, format: ID) => void;
+	// Chat.runHandlers('BattleRanked', this, winnerid, [p1rating, p2rating], [p1id, p2id]);
```
Comment (AnnikaCodes) [inline server/chat.ts:80]: ?

Comment (mia-pi-git) [reply server/chat.ts:80]: Forgot to remove the comment when i was looking at it to write the typedef.

**Comment by AnnikaCodes** [comment]
You'll need to update the `Disconnect` in users.ts (meaning this will also break on hotpatch).

**Comment by AnnikaCodes** [comment]
But in general I like this change!

File: server/chat.ts (hunk 1 above)
Comment (mia-pi-git) [reply server/chat.ts]: Removed with a commit to fix onDisconnect.

**Comment by mia-pi-git** [comment]
> You'll need to update the `Disconnect` in users.ts (meaning this will also break on hotpatch).

To be clear, it won't crash hotpatch, or crash in general. It just won't work.

**Comment by AnnikaCodes** [comment]
Yes, but in my book, breaking handlers is detrimental enough to merit waiting for a restart (especially since this change is mostly useful when refactoring handlers, which can be put off for a few weeks).

**Comment by mia-pi-git** [comment]
Fair. Just wanted to clarify for anyone else reading.

**Review by AnnikaCodes — APPROVED** [review]
This looks good! If you've tested it locally it should be good to merge.

**Pull request merged**